operator/src/main/resources/scripts/traceUtils.sh

```diff
@@ -164,3 +164,107 @@ function exportEffectiveDomainHome() {
   trace "Error: More than one config.xml found at DOMAIN_HOME/config/config.xml and DOMAIN_HOME/*/config/config.xml, DOMAIN_HOME='$DOMAIN_HOME': ${found_configs}. Configure your 'domainHome' setting in your WebLogic Operator Domain resource to reference a single WebLogic domain."
   return 1
 }
+
+
+# versionCmp
+#   Compares two wl versions $1 $2 up to the length of $2
+#     Expects form N.N.N.N
+#     Uses '0' for an N in $1 if $1 is shorter than $2
+#   echo "1"  if v1 >  v2
+#   echo "-1" if v1 <  v2
+#   echo "0"  if v1 == v2
+versionCmp()
+{
+  IFS='.' read -r -a v1_arr <<< "`echo $1`"
+  IFS='.' read -r -a v2_arr <<< "`echo $2`"
+
+  for i in "${!v2_arr[@]}"
+  do
+    [ ${v1_arr[i]:-0} -gt ${v2_arr[i]:-0} ] && echo "1" && return
+    [ ${v1_arr[i]:-0} -lt ${v2_arr[i]:-0} ] && echo "-1" && return
+  done
+  echo "0"
+}
+
+# versionGE
+#   return success if WL v1 >= v2
+versionGE()
+{
+  [ `versionCmp "$1" "$2"` -ge 0 ] && return 0
+  return 1
+}
+
+# versionEQ
+#   return success if v1 == v2
+versionEQ()
+{
+  [ `versionCmp "$1" "$2"` -eq 0 ] && return 0
+  return 1
+}
+
+# hasWebLogicPatches
+#   check for the given patch numbers in the install inventory, 
+#   and return 1 if not found
+#   - if we can't find the install inventory then we 
+#     assume the patch is there...
+#   - we parse the install inventory as this is far faster than
+#     using opatch or weblogic.version
+hasWebLogicPatches()
+{
+  local reg_file=$ORACLE_HOME/inventory/registry.xml
+  [ ! -f $reg_file ] && return 0
+  for pnum in "$@"; do
+    grep --silent "patch-id=\"$1\"" $reg_file || return 1
+  done
+}
+
+# getWebLogicVersion
+#   parse wl version from install inventory
+#   - if we can't get a version number then we return
+#     a high dummy version number that's sufficient
+#     to pass version checks "9999.9999.9999.9999"
```

Comment:
do we log error if unable to find WLS version number?

Reply:
No.  I prefer not to since parsing the inventory file is not a 'standard' way to get the version number and I'm not 100% sure it's reliable.  I think this is OK since checking the version number isn't critical, and we don't want to fail the pod/job or unnecessarily concern end-users if the check itself cannot determine the version number...

Reply:
Yes, agree that failing pod/job for this is not appropriate. What about just logging a message? 
Would users be surprised to see version number "9999.9999.9999.9999" in the trace message "trace "Info: WebLogic version='$cur_wl_ver'. Version check passed. (The Operator requires WebLogic version '${exp_wl_ver}' or higher)."?

Reply:
OK. I'll update to report a different message - something like  "Could not determine WebLogic Version, so assuming that it's correct."

Reply:
I've made the change.  Please resolve if you're fine with it.